backend/src/main/java/io/papermc/hangar/db/dao/internal/versions/HangarVersionsDAO.java

```diff
@@ -94,10 +94,10 @@ WHEN exists(SELECT * FROM pinned_versions piv WHERE piv.version_id = pv.id AND l
                    <endif>)
                    AND
                <endif>
-               lower(p.slug) = lower(:slug) AND
-               pv.version_string = :versionString
+               p.id = :projectId AND
+               pv.id = :versionId
         """)
-    HangarVersion getVersionWithVersionString(String slug, String versionString, @Define boolean canSeeHidden, @Define Long userId);
+    HangarVersion getVersionWithVersionString(long projectId, long versionId, @Define boolean canSeeHidden, @Define Long userId);
```

Comment:
I think the join on projects can be removed here, we only ever use the project id, which is now a param

Reply:
I think the entire method can be removed, as it is basically identical to getVersion now

Reply:
I can just use `pv.project_id` for any places that need to check the project id, right?

Reply:
yes